examples/cis3-nft-sponsored-txs/tests/tests.rs

```diff
@@ -19,11 +26,419 @@ const ACC_INITIAL_BALANCE: Amount = Amount::from_ccd(10000);
+            contract_address: ContractAddress {
+                index:    0,
+                subindex: 0,
+            },
+            entry_point:      OwnedEntrypointName::new_unchecked("updateOperator".into()),
+            nonce:            0,
+            payload:          to_bytes(&payload),
+        },
+    };
+
+    // Update operator with the permit function.
+    let update = chain
+        .contract_update(
+            Signer::with_one_key(),
+            CHARLIE,
+            Address::Account(CHARLIE),
+            Energy::from(10000),
+            UpdateContractPayload {
+                amount:       Amount::zero(),
+                address:      contract_address,
+                receive_name: OwnedReceiveName::new_unchecked("cis3_nft.permit".to_string()),
+                message:      OwnedParameter::from_serial(&permit_update_operator_param)
+                    .expect("Should be a valid inut parameter"),
+            },
+        )
+        .expect("Should be able to update operator with permit");
+
+    // Check that the correct events occurred.
+    let events = update
+        .events()
+        .flat_map(|(_addr, events)| events.iter().map(|e| e.parse().expect("Deserialize event")))
+        .collect::<Vec<Event>>();
+
+    assert_eq!(events, [
+        Event::Cis2Event(Cis2Event::UpdateOperator(UpdateOperatorEvent {
+            update:   OperatorUpdate::Add,
+            owner:    ALICE_ADDR,
+            operator: BOB_ADDR,
+        })),
+        Event::Nonce(NonceEvent {
+            account: ALICE,
+            nonce:   0,
+        })
+    ]);
+
+    // Check operator in state
+    let bob_is_operator_of_alice = operator_of(&chain, contract_address);
+
+    assert_eq!(bob_is_operator_of_alice, OperatorOfQueryResponse(vec![true]));
+}
+
+/// Test permit transfer function. The signature is generated in the test case.
+/// TOKEN_1 is transferred from Alice to Bob.
+#[test]
+fn test_inside_signature_permit_transfer() {
+    let (mut chain, contract_address, _update, keypairs) =
+        initialize_contract_with_alice_tokens(true);
+
+    // Check balances in state.
+    let balance_of_alice_and_bob = get_balances(&chain, contract_address);
+
+    assert_eq!(balance_of_alice_and_bob.0, [TokenAmountU8(1), TokenAmountU8(0)]);
+
+    // Create input parameters for the `permit` transfer function.
+    let transfer = concordium_cis2::Transfer {
+        from:     ALICE_ADDR,
+        to:       Receiver::from_account(BOB),
+        token_id: TOKEN_1,
+        amount:   ContractTokenAmount::from(1),
+        data:     AdditionalData::empty(),
+    };
+    let payload = TransferParams::from(vec![transfer]);
+
+    let mut inner_signature_map = BTreeMap::new();
+    inner_signature_map.insert(0u8, concordium_std::Signature::Ed25519(DUMMY_SIGNATURE));
+
+    let mut signature_map = BTreeMap::new();
+    signature_map.insert(0u8, CredentialSignatures {
+        sigs: inner_signature_map,
+    });
```

Comment:
This seems a bit odd. There is a "dummy" signature that is hardcoded, but the test header claims that signatures are generated as part of the test.

It should be explained why this is done, and what role the dummy signature plays.